Src/IronPython/Modules/_io.cs

```diff
@@ -2797,9 +2798,23 @@ public class TextIOWrapper : _TextIOBase, IEnumerator<object>, IEnumerable<objec
                 mode += '+';
             }
 
-            FileIO fio = fname != null
-                ? new FileIO(context, fname, mode, closefd)
-                : new FileIO(context, fd, mode, closefd);
+            FileIO fio;
+            if (opener != null) {
+                object fdobj = PythonOps.CallWithContext(context, opener, file, mode);
```

Comment:
I don't think passing in the `mode` argument is correct here since the opened expects `flags`. (See https://docs.python.org/3/library/os.html#os.open for info in flags). That being said I'm not exactly sure how the flags value is generated.

Reply:
Good catch.

It looks like `flags` is [created manually](https://github.com/python/cpython/blob/507434fd504f3ebc1da72aa77544edc0d73f136e/Lib/_pyio.py#L1465-L1495).

Reply:
I looked over the test code I had thrown together and realized I had manually set the flags to O_RDONLY in the opener, so I wasn't actually using the flags/mode variable being passed in. Using os.open directly indeed fails.

---

IMHO the cleanest way to go about creating the `flags` int is to create a function which builds the int from `FileMode` and `FileAccess` enums, effectively the opposite of [these functions](https://github.com/IronLanguages/ironpython3/blob/903293eef46c6e8ef764f5b2e35fa5168a0ccdf6/Src/IronPython.Modules/nt.cs#L1753-L1773). This approach would let us model the structure after the CPython implementation, without needing to manually or O_APPEND, O_CREAT, etc. together.

The issue I'm running into is that all of the existing O_* definitions live in `Src/IronPython.Modules/nt.cs`, which is not accessible from `Src/IronPython/Modules/_io.cs`. I'm not sure what the best cross-platform way to handle this is.

What are your thoughts?

Reply:
I'm fine with redefining the constants locally. I believe they come from fcntl.h which isn't Python specific anyway.